solutions/instances/main.tf

```diff
@@ -0,0 +1,170 @@
+  resource_group_name          = var.existing_resource_group == false ? var.resource_group_name : null
+  existing_resource_group_name = var.existing_resource_group == true ? var.resource_group_name : null
+}
+
+module "observability_instance" {
+  source  = "terraform-ibm-modules/observability-instances/ibm"
+  version = "2.11.0"
+  providers = {
+    logdna.at = logdna.at
+    logdna.ld = logdna.ld
+  }
+  region            = var.region
+  resource_group_id = module.resource_group.resource_group_id
+  enable_archive    = var.enable_archive
+  ibmcloud_api_key  = local.archive_api_key
+  # Log Analysis
+  log_analysis_provision           = true
+  log_analysis_instance_name       = var.log_analysis_instance_name
+  log_analysis_plan                = var.log_analysis_plan
+  log_analysis_tags                = var.log_analysis_tags
+  log_analysis_service_endpoints   = var.log_analysis_service_endpoints
+  log_analysis_cos_instance_id     = local.cos_instance_id
+  log_analysis_cos_bucket_name     = local.archive_cos_bucket_name
+  log_analysis_cos_bucket_endpoint = local.archive_cos_bucket_endpoint
+  # IBM Cloud Monitoring
+  cloud_monitoring_provision         = true
+  cloud_monitoring_instance_name     = var.cloud_monitoring_instance_name
+  cloud_monitoring_plan              = var.cloud_monitoring_plan
+  cloud_monitoring_tags              = var.cloud_monitoring_tags
+  cloud_monitoring_service_endpoints = var.cloud_monitoring_service_endpoints
+
+  # Activity Tracker
+  activity_tracker_provision = false
+  cos_targets = [
+    {
+      bucket_name                       = local.cos_target_bucket_name
+      endpoint                          = local.cos_target_bucket_endpoint
+      instance_id                       = local.cos_instance_crn
+      target_region                     = var.cos_region
+      target_name                       = "cos-target"
+      skip_atracker_cos_iam_auth_policy = false
+      service_to_service_enabled        = true
+    }
+  ]
+
+  # Routes
+  activity_tracker_routes = [
+    {
+      route_name = "at-route"
+      locations  = ["*", "global"]
+      target_ids = [
+        module.observability_instance.activity_tracker_targets["cos-target"].id
+      ]
+    }
+  ]
+}
+
+#######################################################################################################################
+# KMS Key
+#######################################################################################################################
+
+# KMS root key for COS bucket
+module "kms" {
+  providers = {
+    ibm = ibm.kms
+  }
+  count                       = var.existing_cos_kms_key_crn != null || var.existing_log_archive_cos_bucket_name != null ? 0 : 1 # no need to create any KMS resources if passing an existing key, or bucket
+  source                      = "terraform-ibm-modules/kms-all-inclusive/ibm"
+  version                     = "4.8.3"
+  resource_group_id           = null # rg only needed if creating KP instance
```

Comment:
you can actually remove this now - in newer kms module, `resource_group_id` defaults to null